stories/core/ExampleComponents/FreeDrawExample.js

```diff
@@ -12,37 +12,49 @@ export default class FreeDrawExample extends PureComponent {
     onDrawBegin: PropTypes.func,
     dataStyle: PropTypes.object,
     theme: PropTypes.object,
+    onDrawEnd: PropTypes.func,
+    allowMultipleShapes: PropTypes.bool,
   }
 
   static defaultProps = {
     theme: {},
     shapes: {},
+    allowMultipleShapes: false,
   }
 
-  constructor(props) {
-    super(props)
-    this.state = {
-      shapes: this.props.shapes,
-      enabled: false,
-    }
+  state = {
+    shapes: this.props.shapes,
+    enabled: this.props.allowMultipleShapes,
   }
 
   @autobind
   handleOnDrawEnd(shape) {
+    const { shapes } = this.state
+
+    if (this.props.onDrawEnd) this.props.onDrawEnd(shape)
+    const shapeId = Object.keys(shapes).length
     this.setState({
-      shapes: shape && { 0: shape },
-      enabled: false,
+      shapes: shape && { ...shapes, [shapeId]: shape },
+      enabled: this.props.allowMultipleShapes,
     })
   }
 
   @autobind
   handleClick() {
-    this.setState({ enabled: !this.state.enabled })
+    const existingShapes = this.props.shapes
+    const { enabled, shapes: _ } = this.state
```

Comment:
`_` usually denotes you aren't using that prop. Cant' we just rename it to something else? I'd prefer the code below to be readable.